src/components/TransactionFlow/TxReceipt.tsx

```diff
@@ -162,20 +178,50 @@ const TxReceipt = ({
   }, []);
 
   const assetRate = (() => {
-    if (displayTxReceipt && path(['asset'], displayTxReceipt)) {
-      return getAssetRate(displayTxReceipt.asset);
+    if (displayTxReceipt && path(['valueTransfers'], displayTxReceipt) && displayTxReceipt.valueTransfers.length > 0) {
+      return getAssetRate(displayTxReceipt.valueTransfers[0].asset);
     } else {
       return getAssetRate(txConfig.asset);
     }
-  })();
-
-  const baseAssetRate = (() => {
-    if (displayTxReceipt && path(['baseAsset'], displayTxReceipt)) {
-      return getAssetRate(displayTxReceipt.baseAsset);
-    } else {
-      return getAssetRate(txConfig.baseAsset);
+  })
+
+  const baseAssetRate = getAssetRate(txConfig.baseAsset);
```

Comment:
Is it safe to move this out?

Reply:
I'm not sure why it wouldn't be. There's no reason for it to be based off of `displayTxReceipt` instead of `txConfig`. They both pull from the same place.